app/server/websocketServer.php

```diff
@@ -0,0 +1,607 @@
+<?php
+
+
+require_once('WebSocketUser.php');
+
+abstract class WebSocketServer {
+
+  protected $userClass = 'WebSocketUser'; // redefine this if you want a custom user class.  The custom user class should inherit from WebSocketUser.
+  protected $maxBufferSize;        
+  protected $master;
+  protected $sockets                              = array();
+  protected $users                                = array();
+  protected $heldMessages                         = array();
+  protected $interactive                          = true;
+  protected $headerOriginRequired                 = false;
+  protected $headerSecWebSocketProtocolRequired   = false;
+  protected $headerSecWebSocketExtensionsRequired = false;
+
+  function __construct($addr, $port, $bufferLength = 2048) {
+    $this->maxBufferSize = $bufferLength;
+    $this->master = socket_create(AF_INET, SOCK_STREAM, SOL_TCP)  or die("Failed: socket_create()");
```

Comment:
ich weiß die ist oldschool aber wie wär's mit exceptions ? :D sind irgendwie besser zum tracken :) vor allem kannst diese evtl mit monolog oder einem anderen log stream einfach in kibana oder ein tool deiner wahl streamen :)